daemon/logger/syslog/syslog.go

```diff
@@ -0,0 +1,61 @@
+package syslog
+
+import (
+	"log/syslog"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/docker/docker/daemon/logger"
+)
+
+type Syslog struct {
+	writer *syslog.Writer
+	tag    string
+	mu     sync.Mutex
+}
+
+func New(tag string) (logger.Logger, error) {
+	return &Syslog{
+		writer: nil,
+		tag:    tag,
+	}, nil
+}
+
+func (s *Syslog) Log(msg *logger.Message) error {
+	s.mu.Lock()
```

Comment:
@LK4D4 you're right. It reconnects if there is an error after the first connection. If there was no first connection, it wouldn't be able to reconnect. https://golang.org/src/log/syslog/syslog.go Line #145